### PR TITLE
Fix traceanalyzer NLID unittest

### PR DESCRIPTION
**Pull request by klyr**

### CODE REVIEW

File: backend/pkg/modules/internal/traceanalyzer/nlid/nlid_test.go [hunk 1]
```diff
@@ -16,21 +16,48 @@
...
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+
+	return true
+}
```
[the remark this answers is not in the document]
Comment (klyr) [reply backend/pkg/modules/internal/traceanalyzer/nlid/nlid_test.go:51]: yes: https://github.com/openclarity/apiclarity/pull/179/files/98ca3d0927e533b8529e7bd6768215a6947defa8#diff-14e723a397271eb80fb858946efe7d093a4a94cd5c4cc213e788c8ce34bc842dR25

[the remark this answers is not in the document]
Comment (klyr) [reply backend/pkg/modules/internal/traceanalyzer/nlid/nlid_test.go:51]: I mean, sorting then comparing is the same as comparing unsorted array

[the remark this answers is not in the document]
Comment (aduminuc) [reply backend/pkg/modules/internal/traceanalyzer/nlid/nlid_test.go:51]: where is the sorting?

[the remark this answers is not in the document]
Comment (aduminuc) [reply backend/pkg/modules/internal/traceanalyzer/nlid/nlid_test.go:51]: ok, I meant using sort.Slice , but it's equivalent, was just a curiosity

**Pull request merged**